include/zephyr/mgmt/mcumgr/smp/mcumgr_client.h

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef H_MGMT_MCUMGR_CLIENT_
+#define H_MGMT_MCUMGR_CLIENT_
+
+#include <inttypes.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @brief MCUMGR Image Group response types.
+ */
+enum mcumgr_image_rsp_t {
+	/** Image state write or read response */
+	MGMT_IMAGE_STATE_RSP = 0,
+
+	/** Image Upload response*/
+	MGMT_IMAGE_UPLOAD_RSP,
+
+	/** Secondary image erase response */
+	MGMT_IMAGE_ERASE_RSP
+};
+
+/**
+ * @brief MCUMGR OS Group response types.
+ */
+enum mcumgr_os_rsp_t {
+	/** OS response */
+	MGMT_OS_ECHO_RSP = 0,
```

Comment:
All of this needs splitting up, a "client" should only have the functionality needed for the client, i.e. no command processing, command processing should be moved to independent files (for each group e.g. fs, os, img, etc.)